aodncore/util/s3_util.py

```diff
@@ -0,0 +1,64 @@
+from urllib.parse import urlparse
+import boto3
```

Comment:
Some functionality for handling objects on S3 is already implemented as an [`S3StorageBroker`](https://github.com/aodn/python-aodncore/blob/0fe87c95681ae535fd6eabcae352cb806dcfb0f7/aodncore/pipeline/storage.py#L304) class that suports query, download, upload and delete operations. Could that be reused instead of duplicating the functionality here?